tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4275 @@
+  } else if (OB_FAIL(ObAdminBackupValidationUtil::read_backup_info_file(
+                 full_path, *backup_storage_info, locality_info_desc, ctx_))) {
+    STORAGE_LOG(WARN, "failed to read backup info file", K(ret), K(full_path));
+  } else {
+    STORAGE_LOG(INFO, "succeed to check locality file", K(ret), K(full_path));
+  }
+  if (OB_FAIL(ret)) {
+    ctx_->go_abort(full_path.get_ptr(), "locality file not correct");
+  }
+  return ret;
+}
+int ObAdminBackupSetMetaValidationTask::collect_tenant_ls_meta_info_()
+{
+  int ret = OB_SUCCESS;
+  int64_t backup_set_id = -1;
+  share::ObBackupPath full_path;
+  ObAdminBackupSetAttr *backup_set_attr = nullptr;
+  storage::ObBackupDataStore *backup_set_store = nullptr;
+  share::ObBackupStorageInfo *backup_storage_info = nullptr;
+  storage::ObBackupLSMetaInfosDesc ls_meta_infos_desc;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_FAIL(ctx_->processing_backup_set_id_array_.at(task_id_, backup_set_id))) {
+    STORAGE_LOG(WARN, "failed to get backup set id", K(ret), K(task_id_));
+  } else if (OB_FAIL(ctx_->get_backup_set_attr(backup_set_id, backup_set_attr))) {
+    STORAGE_LOG(WARN, "failed to get backup set store", K(ret), K(backup_set_id));
+  } else if (OB_ISNULL(backup_set_attr)) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "backup data store is null", K(ret));
+  } else if (OB_ISNULL(backup_set_store = backup_set_attr->backup_set_store_)) {
+  } else if (OB_ISNULL(backup_storage_info
+                       = backup_set_store->get_backup_set_dest().get_storage_info())) {
+    ret = OB_ERR_UNEXPECTED;
+    STORAGE_LOG(WARN, "backup storage info is null", K(ret));
+  } else if (OB_FAIL(share::ObBackupPathUtil::get_ls_meta_infos_path(
+                 backup_set_store->get_backup_set_dest(), full_path))) {
+    STORAGE_LOG(WARN, "failed to get locality info path", K(ret), K(full_path));
+  } else if (OB_FAIL(ObAdminBackupValidationUtil::read_backup_info_file(
+                 full_path, *backup_storage_info, ls_meta_infos_desc, ctx_))) {
+    STORAGE_LOG(WARN, "failed to read backup info file", K(ret), K(full_path));
+  } else {
+    STORAGE_LOG(INFO, "succeed to collect tenant ls meta info", K(ret), K(full_path));
+    FOREACH_X(ls_meta_package, ls_meta_infos_desc.ls_meta_packages_, OB_SUCC(ret))
+    {
+      const ObLSMeta &ls_meta = ls_meta_package->ls_meta_;
+      ObAdminLSAttr *ls_attr = nullptr;
+      if (OB_FAIL(ctx_->add_ls(backup_set_id, ls_meta.ls_id_))) {
+        STORAGE_LOG(WARN, "failed to add ls", K(ret), K(ls_meta.ls_id_));
+      } else if (OB_FAIL(ctx_->get_ls_attr(backup_set_id, ls_meta.ls_id_, ls_attr))) {
+        STORAGE_LOG(WARN, "failed to get ls attr", K(ret), K(ls_meta.ls_id_));
+      } else {
+        ls_attr->ls_meta_package_ = *ls_meta_package;
+      }
+    }
+  }
+
+  return ret;
+}
+int ObAdminBackupSetMetaValidationTask::collect_inner_tablet_meta_index_()
```

Comment:
do consider split the function logic into smaller pieces

Reply:
split out an inner_assign_meta_index_to_tablet_attr_